src/lib.rs

```diff
@@ -459,6 +608,83 @@ macro_rules! destroy_with_runtime {
     };
 }
 
+#[macro_export]
+macro_rules! export_queued_op {
+    ($name: ident, $response: ty, $builder: expr, $($v:ident: $t:ty),+) => {
+        #[no_mangle]
+        pub extern "C" fn $name($($v: $t),+, config: *const RawConfig, response: *mut $response, handle: *const c_void) -> CResult {
+            let response = unsafe { ResponseGuard::new(response, handle) };
+            let config = unsafe { & (*config) };
+            let req_result: Result<Request, (ResponseGuard<$response>, anyhow::Error)> = $builder(config, response);
+            let req = match req_result {
+                Ok(req) => req,
+                Err((response, e)) => {
+                    response.into_error(e);
+                    return CResult::Error;
+                }
+            };
+            match SQ.get() {
+                Some(sq) => {
+                    match sq.try_send(req) {
+                        Ok(_) => CResult::Ok,
+                        Err(flume::TrySendError::Full(req)) => {
+                            req.into_error("object_store_ffi internal channel full, backoff");
+                            CResult::Backoff
+                        }
+                        Err(flume::TrySendError::Disconnected(req)) => {
+                            req.into_error("object_store_ffi internal channel closed (may be missing initialization)");
+                            CResult::Error
+                        }
+                    }
+                }
+                None => {
+                    req.into_error("object_store_ffi internal channel closed (may be missing initialization)");
+                    return CResult::Error;
+                }
+            }
+        }
+
+    };
+}
+
+// TODO use macro for exporting runtime operations
```

Comment:
still TODO?

Reply:
Yes, we are using the macro for exporting queued operations (`export_queued_op`) but still not using this one for runtime operations